### ISSUE TITLE
Create ~/.codalab/misc_temp automatically

### ISSUE THREAD

**Issue by kashizui** [codalab/codalab-worksheets#359]
I get this error when running `cl rest-server` if `~/.codalab/misc_temp` directory doesn't exist yet:

```
➜  cl rest-server                                                                                                                                             16-02-28 17:44:02
Bottle v0.12.9 server starting up (using GunicornServer(workers=1, worker_class='gevent'))...
Listening on http://127.0.0.1:8080/
Hit Ctrl-C to quit.

[2016-02-28 17:44:15 -0800] [32790] [INFO] Starting gunicorn 19.4.5
[2016-02-28 17:44:15 -0800] [32790] [INFO] Listening at: http://127.0.0.1:8080 (32790)
[2016-02-28 17:44:15 -0800] [32790] [INFO] Using worker: gevent
[2016-02-28 17:44:15 -0800] [32790] [INFO] Unhandled exception in main loop:
Traceback (most recent call last):
  File "/Users/sckoo/Developer/codalab-cli/venv/lib/python2.7/site-packages/gunicorn/arbiter.py", line 180, in run
    self.manage_workers()
  File "/Users/sckoo/Developer/codalab-cli/venv/lib/python2.7/site-packages/gunicorn/arbiter.py", line 482, in manage_workers
    self.spawn_workers()
  File "/Users/sckoo/Developer/codalab-cli/venv/lib/python2.7/site-packages/gunicorn/arbiter.py", line 548, in spawn_workers
    self.spawn_worker()
  File "/Users/sckoo/Developer/codalab-cli/venv/lib/python2.7/site-packages/gunicorn/arbiter.py", line 502, in spawn_worker
    self.cfg, self.log)
  File "/Users/sckoo/Developer/codalab-cli/venv/lib/python2.7/site-packages/gunicorn/workers/async.py", line 24, in __init__
    super(AsyncWorker, self).__init__(*args, **kwargs)
  File "/Users/sckoo/Developer/codalab-cli/venv/lib/python2.7/site-packages/gunicorn/workers/base.py", line 55, in __init__
    self.tmp = WorkerTmp(cfg)
  File "/Users/sckoo/Developer/codalab-cli/venv/lib/python2.7/site-packages/gunicorn/workers/workertmp.py", line 23, in __init__
    fd, name = tempfile.mkstemp(prefix="wgunicorn-", dir=fdir)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/tempfile.py", line 308, in mkstemp
    return _mkstemp_inner(dir, prefix, suffix, flags)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/tempfile.py", line 239, in _mkstemp_inner
    fd = _os.open(file, flags, 0600)
OSError: [Errno 2] No such file or directory: '/Users/sckoo/.codalab/misc_temp/wgunicorn-XMcSNO'
```

**Comment by a10y** [comment]
This should already be happening, did you try clearing your config and state.json when you updated?

**Comment by percyliang** [comment]
We can't expect people to just clear their config if they update the CLI...

**Comment by a10y** [comment]
Realize now why this is happening: it's because if all you run is `cl rest-server`, then you never actually cause the bundle_store() method on CodaLabManager to get run. You need to run a command like `cl ls` to generate the directory. The other way we can get around this is have CodaLabManager call `self.bundle_store()` immediately when it is instantiated.

**Comment by a10y** [comment]
faadf65 implements the above solution.

**Comment by kashizui** [comment]
Fixed by faadf65
